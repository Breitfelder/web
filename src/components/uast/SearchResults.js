import React, { Component } from 'react';
import styled from 'styled-components';
import Node from './Node';

const NotFound = styled.div`
  width: 100%;
  padding: 1rem 0;
  text-align: center;
`;

export default class SearchResults extends Component {
  render() {
    const { resultIds, onNodeSelected } = this.props;

    if (!resultIds || !resultIds.length) {
      return <NotFound>Nothing found</NotFound>;
    }

    return (
      <div>
        {resultIds.map(id => {
          return <Node key={id} id={id} onNodeSelected={onNodeSelected} />;
        })}
      </div>
    );
  }
}
