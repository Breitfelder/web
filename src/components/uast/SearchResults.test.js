import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import SearchResults from './SearchResults';

const mockStore = configureStore();

const shouldMatchSnapshot = comp => {
  const wrapper = renderer.create(comp);
  expect(wrapper.toJSON()).toMatchStyledComponentsSnapshot();
};

describe('SearchResults', () => {
  it('should render children nodes', () => {
    const store = mockStore({
      code: {
        ast: {
          2: { id: 2, InternalType: 'Child1' },
          3: { id: 3, InternalType: 'Child2' },
        },
      },
      options: { showLocations: false },
    });
    shouldMatchSnapshot(
      <Provider store={store}>
        <SearchResults resultIds={[2, 3]} />
      </Provider>
    );
  });

  it('should render not found', () => {
    shouldMatchSnapshot(<SearchResults resultIds={[]} />);
  });
});
