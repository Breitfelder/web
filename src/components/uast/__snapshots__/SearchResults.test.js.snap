// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults should render children nodes 1`] = `
.jQbgOF {
  margin-left: 20px;
  min-width: 400px;
  background: none;
}

.hRiJmB {
  min-height: 29px;
}

.hRiJmB::before {
  content: '';
  width: 15px;
  display: inline-block;
}

.hbfmnM {
  min-height: 29px;
  cursor: pointer;
}

.hbfmnM::before {
  content: '';
  width: 15px;
  display: inline-block;
}

.hbfmnM::before {
  content: '+';
  color: #EB7422;
}

.hbfmnM:hover > span {
  border-bottom: 1px dashed #B1B1B1;
}

.bUwxHt {
  display: none;
  border-left: 1px solid #dad7d0;
  margin-left: 4px;
}

.gaIEnP {
  display: inline;
  font-family: Hack, monospace;
  color: #B1B1B1;
}

.HnGWi {
  color: #F2A83F;
}

.HnGWi::before {
  content: '\\27';
}

.HnGWi::after {
  content: '\\27';
}

.iOQEqP {
  margin-right: 5px;
}

.iOQEqP::after {
  content: ':';
  color: #B1B1B1;
}

<div>
  <div
    className="sc-bdVaJa jQbgOF"
    onMouseMove={[Function]}
  >
    <div
      className="sc-htpNat hbfmnM"
      onClick={[Function]}
    >
      <summary
        className="sc-ifAKCX gaIEnP"
      >
        Node
      </summary>
    </div>
    <div
      className="sc-bxivhb bUwxHt"
    >
      <div
        className="sc-bdVaJa jQbgOF"
      >
        <div
          className="sc-bwzfXH hRiJmB"
        >
          <span
            className="sc-bZQynM iOQEqP"
          >
            internal_type
          </span>
          <span
            className="sc-EHOje HnGWi"
            type="string"
          >
            Child1
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="sc-bdVaJa jQbgOF"
    onMouseMove={[Function]}
  >
    <div
      className="sc-htpNat hbfmnM"
      onClick={[Function]}
    >
      <summary
        className="sc-ifAKCX gaIEnP"
      >
        Node
      </summary>
    </div>
    <div
      className="sc-bxivhb bUwxHt"
    >
      <div
        className="sc-bdVaJa jQbgOF"
      >
        <div
          className="sc-bwzfXH hRiJmB"
        >
          <span
            className="sc-bZQynM iOQEqP"
          >
            internal_type
          </span>
          <span
            className="sc-EHOje HnGWi"
            type="string"
          >
            Child2
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`SearchResults should render not found 1`] = `
.kqrAPn {
  width: 100%;
  padding: 1rem 0;
  -webkit-text-align: center;
  text-align: center;
}

<div
  className="sc-gzVnrw kqrAPn"
>
  Nothing found
</div>
`;
